build/docker/zap-api-scan.py

```diff
@@ -0,0 +1,685 @@
+#!/usr/bin/env python
+# Zed Attack Proxy (ZAP) and its related class files.
+#
+# ZAP is an HTTP/HTTPS proxy for assessing web application security.
+#
+# Copyright 2017 ZAP Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs a full scan against an API defined by OpenAPI/Swagger or SOAP
+# using ZAP
+#
+# It can either be run 'standalone', in which case depends on
+# https://pypi.python.org/pypi/python-owasp-zap-v2.4 and Docker, or it can be run
+# inside one of the ZAP docker containers. It automatically detects if it is
+# running in docker so the parameters are the same.
+#
+# It currently support APIS defined by:
+#	OpenAPI/Swagger URL
+#	OpenAPI/Swagger file
+#	SOAP URL
+#	SOAP File
+# It will exit with codes of:
+#	0:	Success
+#	1:	At least 1 FAIL
+#	2:	At least one WARN and no FAILs
+#	3:	Any other failure
+# By default all alerts found by ZAP will be treated as WARNings.
+# You can use the -c or -u parameters to specify a configuration file to override
+# this.
+# You can generate a template configuration file using the -g parameter. You will
+# then need to change 'WARN' to 'FAIL', 'INFO' or 'IGNORE' for the rules you want
+# to be handled differently.
+# You can also add your own messages for the rules by appending them after a tab
+# at the end of each line.
+# The active scan rules run are currently hardcoded in the API-Minimal.policy file
+# the plan is to make these configurable in a future version of this script.
+
+import getopt
+import json
+import logging
+import os
+import os.path
+import re
+import socket
+import subprocess
+import sys
+import time
+import traceback
+import urllib2
+from datetime import datetime
+from random import randint
+from zapv2 import ZAPv2
+
+timeout = 120
+config_dict = {}
+config_msg = {}
+out_of_scope_dict = {}
+running_in_docker = os.path.exists('/.dockerenv')
+levels = ["PASS", "IGNORE", "INFO", "WARN", "FAIL"]
+min_level = 0
+
+# Scan rules that aren't really relevant, eg the examples rules in the alpha set
+blacklist = ['-1', '50003', '60000', '60001']
+
+# Scan rules that are being addressed
+in_progress_issues = {}
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s %(message)s')
+# Hide "Starting new HTTP connection" messages
+logging.getLogger("requests").setLevel(logging.WARNING)
+
+def usage():
+    print ('Usage: zap-baseline.py -t <target> [options]')
```

Comment:
:+1:

I think this should be changed to `'Usage: zap-api-scan.py` rather than still mentioning `zap-baseline.py`. Also, even with the manual help text, it might be easier to maintain if it was one multiline string, i.e.
```python
print("""Usage: zap-api-scan.py -t <target> [options]
    -t target         target API definition, currently only an OpenAPI URL, eg https://www.example.com/openapi.json
    -f format         either openapi or soap
...etc....
""")
```

Which could also be done in both.